app/models/task.rb

```diff
@@ -1,17 +1,26 @@
 # frozen_string_literal: true
 
 class Task < ApplicationRecord
+  SIZES = %w[small medium large].freeze
+
   belongs_to :board, optional: true
   belongs_to :owner, class_name: "User", optional: true
+
+  before_validation :nillify_size
   before_save :update_approved
 
   validates :title, presence: true
+  validates :size, inclusion: { in: SIZES }, allow_nil: true
 
   scope :active, -> { where(completed_at: nil) }
   scope :recently_completed, -> { where("completed_at > ?", 1.week.ago) }
 
   private
 
+  def nillify_size
+    self.size = nil if size&.empty? # rubocop:disable Lint/SafeNavigationWithEmpty
```

Comment:
Don't use empty use blank. Empty is for array like things. Also it raises when called on `nil` blank doesn't so you don't need safe navigation. Rubocop is not wrong here